### ISSUE TITLE
Quick pick: Support alternate representation of French/European characters

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#196733]
This should also match `tache`:

![Image](https://github.com/microsoft/vscode/assets/2193314/4f0df7f1-b411-420d-9cfd-28a8b6c045f7)

See https://github.com/microsoft/vscode/issues/196203 which this would be built upon.

cc @aiday-mar

**Comment by aiday-mar** [comment]
For Russian, I would say the following are similar, and can be interchanged for quick typing or by mistake

ё, е
и, й
щ, ш
ъ, ь

**Comment by rzhao271** [comment]
For Chinese, it seems like the IDE already suggests English words for you
I'm unable to get a good example because the IME on my Windows device is busted and Google Pinyin doesn't seem to exist as a standalone app anymore, so here's an example picture of a mobile IME suggesting the English string "yk" along with certain characters:

![Cangjie mobile example](https://2.bp.blogspot.com/-8H0EjuN0aC0/Tw6uD1SOVdI/AAAAAAAAdGM/M_DXwlbpfow/s1600/1.jpg)

**Comment by Tyriar** [comment]
@rzhao271 I think for Chinese and Japanese Kanji (Hiragana and Katakana could work) this isn't really possible due to the number of characters. It would make sense to translate input characters in a desktop Chinese IME into the qwerty alternatives, in case you accidentally started typing when in the wrong keyboard mode, but again we can only really map a handful of characters and/or a unicode code range.

**Comment by rzhao271** [comment]
What I'm saying is that Chinese and Japanese should be out of scope anyway with a good IME because all Chinese IMEs I used suggest English words for you if you type out an English word even while in Chinese input mode.

**Comment by TylerLeonhardt** [comment]
Calling out what @jrieken mentioned in slack:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#options

We may be able to use this for the simpler cases

**Comment by Tyriar** [comment]
Yep I think Collator will cover the latin cases, not sure about Cyrillic

![image](https://github.com/microsoft/vscode/assets/2193314/0dcbbae9-964a-45be-b1d5-87d07d44b6f2)
![image](https://github.com/microsoft/vscode/assets/2193314/b502954f-fab4-4215-905b-84efc40ec854)

**Comment by rzhao271** [comment]
For Russian, I've heard of ё being written as just е, but I'm unsure about the rest.
I'm unsure if it's because of the https://en.wikipedia.org/wiki/JCUKEN layout and how ё is way off in the corner. 
`'ё'.localeCompare('е', undefined, { sensitivity: 'accent' });` returns 1 and `'ё'.localeCompare('е', 'ru', { sensitivity: 'accent' });` also returns 1, though.

**Comment by dmitrivMS** [comment]
> Yep I think Collator will cover the latin cases, not sure about Cyrillic
> 
> ![image](https://github.com/microsoft/vscode/assets/2193314/0dcbbae9-964a-45be-b1d5-87d07d44b6f2) ![image](https://github.com/microsoft/vscode/assets/2193314/b502954f-fab4-4215-905b-84efc40ec854)

Cyrillic is covered as well.

**Comment by dmitrivMS** [comment]
Testing: Using display languages that include characters with accents (European languages), search for words that include accents in command picker (Ctrl+Shift+P), verify that picker shows commands with names including accents, verify that word highlighting works as normal (selected text matches search terms etc).
